Analysis/DEEP2_R23_O32/Binning/N_Bins_Grid_Analysis.py

```diff
@@ -73,71 +84,91 @@ def n_times_binned(fitspath, dataset, pdf_pages, outfile, n_split, R23, O32, SNR
+            O32_minimum[ii, jj] = O32_values_perbin[0]
 
-            # These two give the median R23 and O32 value for each bin
-            O32_median[ii, jj] = np.median(O32_values_perbin)
+            # First two map median R23 and O32 measure to all individual spectra in bin
+            # Second two give the median R23 and O32 value for each bin
+            R23_medall[N_bin_idx] = np.median(R23[N_bin_idx])
+            O32_medall[N_bin_idx] = np.median(O32_values_perbin)
             R23_median[ii, jj] = np.median(R23[N_bin_idx])
+            O32_median[ii, jj] = np.median(O32_values_perbin)
+
+            # First two map maximum R23 and O32 measure to all individual spectra in bin
+            # Second two give the maximum R23 and O32 value for each bin
+            R23_maxall[N_bin_idx] = np.max(R23[N_bin_idx])
+            O32_maxall[N_bin_idx] = np.max(O32[N_bin_idx])
+            R23_max[ii, jj] = np.max(R23[N_bin_idx])
+            O32_max[ii, jj] = np.max(O32[N_bin_idx])
 
             # Gives the number of galaxies in each bin
-            Number_inbin[ii, jj] += len(O32_values_perbin)
+            number_inbin[ii, jj] += len(O32_values_perbin)
 
-            # Gives the index for the spectra in each bin and is used later loop over and get the galaxies
+            # Gives the index (location numbers) for the spectra in each bin
+            # and is used later loop over and get the galaxies
             locator[ii, jj] = N_bin_idx
 
-            # Gives the bin number for each spectra
-            Bin_number[N_bin_idx] = (ii*n_split)+jj
-
+            # Maps average R23 measure to all individual spectra in bin
+            R23_avgall[N_bin_idx] = np.average(R23[N_bin_idx])
             # Gives the average R23 value for each bin
-            xBar[(ii*n_split)+jj] = np.log10(np.average(R23[N_bin_idx]))
+            xBar[(ii*n_split)+jj] = np.average(R23[N_bin_idx])
 
+            # Maps average O32 measure to all individual spectra in bin
+            O32_avgall[N_bin_idx] = np.average(O32_values_perbin)
             # Gives the average O32 value for each bin
-            yBar[(ii*n_split)+jj] = np.log10(np.average(O32_values_perbin))
+            yBar[(ii*n_split)+jj] = np.average(O32_values_perbin)
 
             # Gives the number of galaxies in each bin
             area[(ii*n_split)+jj] = len(O32_values_perbin)
 
             # Now we can shift our window over for the next split like the counting index the Stacking code
             startIdx = endIdx
-            endIdx = startIdx+subbin_arr[jj]+1
+            endIdx = startIdx+subbin_arr[jj] + 1
+
+    O32_lowlimit = O32_minimum.reshape(n_split * n_bins)  # Used to be O32_grids
+    R23_lowlimit = R23_minimum.reshape(n_split * n_bins)  # Used to be R23_grids
+    O32_medians = O32_median.reshape(n_split * n_bins)
+    R23_medians = R23_median.reshape(n_split * n_bins)
+    R23_maxval = R23_max.reshape(n_split * n_bins)
+    O32_maxval = O32_max.reshape(n_split * n_bins)
```

Comment:
It might be nice to use a variable like `n_bins_total = n_split * n_bins`.

Not crucial though.

Reply:
Fixed